roles/create_mce_operator/tasks/main.yml

```diff
@@ -0,0 +1,184 @@
+    definition:
+      apiVersion: multicluster.openshift.io/v1
+      kind: MultiClusterEngine
+      metadata:
+        name: multiclusterengine-sample
+      spec:
+        availabilityConfig: High
+        overrides:
+          components:
+            - enabled: true
+              name: assisted-service
+            - enabled: true
+              name: cluster-lifecycle
+            - enabled: true
+              name: cluster-manager
+            - enabled: true
+              name: discovery
+            - enabled: true
+              name: hive
+            - enabled: true
+              name: server-foundation
+            - enabled: true
+              name: cluster-proxy-addon
+            - enabled: false
+              name: managedserviceaccount-preview
+            - enabled: false
+              name: hypershift-preview
+            - enabled: true
+              name: console-mce
+        targetNamespace: multicluster-engine
+
+- name: Wait until MCE Pods are running
+  kubernetes.core.k8s_info:
+    kind: Pod
+    namespace: multicluster-engine
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+  register: pod_list_2
+  vars:
+    filter_str: '[].{"name": metadata.name, "status": status.phase}'
+  until: pod_list_2.resources | json_query(filter_str) | rejectattr('name', 'in', old_pod_names) | map(attribute='status') | unique | list == ["Running"]
+  retries: 20
+  delay: 10
+
+- name: Patch the provisioning-configuration to enable CIM service
+  kubernetes.core.k8s:
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    state: present
+    api_version: metal3.io/v1alpha1
+    kind: provisioning
+    name: provisioning-configuration
+    definition:
+      spec:
+        watchAllNamespaces: true
+
+- name: Get MCE Pods list
+  kubernetes.core.k8s_info:
+    kind: Pod
+    namespace: multicluster-engine
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+  register: pod_list
+
+- ansible.builtin.set_fact:
+    old_pod_names: "{{ pod_list.resources | json_query('[].metadata.name')}}"
+
+- ansible.builtin.debug:
+    msg: "{{ old_pod_names }}"
+
+- name:  Create the AgentServiceConfig
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    definition:
+      apiVersion: agent-install.openshift.io/v1beta1
+      kind: AgentServiceConfig
+      metadata:
+       name: agent
+      spec:
+        databaseStorage:
+          storageClassName: ocs-storagecluster-cephfs
+          accessModes:
+          - ReadWriteOnce
+          resources:
+            requests:
+              storage: 20Gi
+        filesystemStorage:
+          storageClassName: ocs-storagecluster-cephfs
+          accessModes:
+          - ReadWriteOnce
+          resources:
+            requests:
+              storage: 10Gi
+        imageStorage:
+          storageClassName: ocs-storagecluster-cephfs

```

Comment:
Are these only `cephfs`? What if someone is using LSO?

Reply:
I need to think about this one a bit.

Reply:
storage class check is added